srcCs/PerftestTransport.cs

```diff
@@ -979,19 +1023,248 @@ private void ConfigureSecurityFiles(DDS.DomainParticipantQos qos)
+
+            // /*
+            //  * If we specify -scan, then we are interested in the highest size.
+            //  * Since the vector for scan is sorted, that number should be the last
+            //  * element.
+            //  */
+            // if (_PM->is_set("scan")) {
+            //     const std::vector<unsigned long long> scanList =
+            //             _PM->get_vector<unsigned long long>("scan");
+            //     datalen = scanList[scanList.size() - 1];
+            // }
+
+            ulong parentMsgSizeMax = minimumMessageSizeMax;
+            bool messageSizeMaxSet = false;
+
+            /*
+            * If the property defining the message_size_max for shared memory is not
+            * set and we are using exclusively SHMEM, then we will calculate
+            * automatically the message_size_max to accomodate the sample in a single
+            * packet and avoid fragmentation.
+            */
 
-            // Number of messages that can be buffered in the receive queue.
-            int receivedMessageCountMax = 2 * 1024 * 1024 / (int) dataLen;
-            if (receivedMessageCountMax < 1) {
-                receivedMessageCountMax = 1;
+            if (propertyValue != null)
+            {
+                Console.Out.WriteLine("1");
+                parentMsgSizeMax = Convert.ToUInt64(propertyValue.value);
+                messageSizeMaxSet = true;
             }
+            else
+            {
+                if (qos.transport_builtin.mask == (int) TransportBuiltinKind.TRANSPORTBUILTIN_SHMEM)
+                {
+                    if ((dataLen + MESSAGE_OVERHEAD_BYTES) > parentMsgSizeMax)
+                    {
+                        parentMsgSizeMax = dataLen + MESSAGE_OVERHEAD_BYTES;
+                        minimumMessageSizeMax = parentMsgSizeMax;
+                    }
+                }
 
-            DDS.PropertyQosPolicyHelper.add_property(
+                PropertyQosPolicyHelper.assert_property(
+                        qos.property_qos,
+                        "dds.transport.shmem.builtin.parent.message_size_max",
+                        parentMsgSizeMax.ToString(),
+                        false);
+                loggingString += "\tSHMEM message_size_max: " + parentMsgSizeMax + "\n";
+            }
+
+            ulong maxBufferSize = Math.Max(121634816 /* 116MB */, parentMsgSizeMax);
+
+            if (!messageSizeMaxSet &&
+                    parentMsgSizeMax > maxBufferSize)
+            {
+                parentMsgSizeMax = maxBufferSize;
+
+                minimumMessageSizeMax = parentMsgSizeMax;
+
+                PropertyQosPolicyHelper.assert_property(
+                    qos.property_qos,
+                    "dds.transport.shmem.builtin.parent.message_size_max",
+                    maxBufferSize.ToString(),
+                    false);
+                loggingString += "\tSHMEM message_size_max: " + parentMsgSizeMax + "\n";
+
+            }
+
+            // /*
```

Comment:
Why we have this as a double comment?
// /*?

Reply:
I think it was because I added the code from C++ and I started translating things to C#. Then I forgot to fix the comments.